internal/plugin/connectors/tcp/mssql/mock/options.go

```diff
@@ -0,0 +1,103 @@
+package mock
+
+import (
+	"context"
+	"net"
+
+	logmock "github.com/cyberark/secretless-broker/pkg/secretless/log/mock"
+
+	"github.com/cyberark/secretless-broker/internal/plugin/connectors/tcp/mssql/types"
+	mssql "github.com/denisenkom/go-mssqldb"
+)
+
+// MSSQLConnectorCtorOptions represents options when creating a MSSQLConnectorCtor. This
+// makes it possible to customize some key values in a Connect double:
+// func Connect(...) (net.Conn, error)
+type MSSQLConnectorCtorOptions struct {
+	// Backend connection returned from the call to Connect
+	BackendConn net.Conn
+	// Error returned from the call to Connect
+	Err error
+	// PreloginResponse from the server passed to interceptor during Connect
+	ServerPreloginResponse map[uint8][]byte
+	// Pointer to unload client login request to, taken from interceptor during Connect
+	ClientLoginRequestPtr **mssql.LoginRequest
```

Comment:
There should be no references to `mssql` in any mocks.  This is a very important invariant to maintain.